src/components/AppLayout/NavHeader.tsx

```diff
@@ -40,30 +40,50 @@ interface Props {
 
 const NavHeader = ({ className, sidedrawer, setSidedrawer, previousRoute } : Props) => {
 	const { network } = useNetworkContext();
-	const currentUser = useUserDetailsContext();
+	const { govType, username, setUserDetailsContextState } = useUserDetailsContext();
 	const router = useRouter();
-	const { pathname, query } = router;
-	const { username } = currentUser;
 	const [open, setOpen] = useState(false);
 	const [openLogin,setLoginOpen]=useState<boolean>(false);
 	const [openSignup,setSignupOpen]=useState<boolean>(false);
-
-	const isGov2Route: boolean = checkGov2Route(pathname, query, previousRoute, network);
 	const isClicked = useRef(false);
 
+	const setGovTypeToContext = (govType: EGovType) => {
+		setUserDetailsContextState((prev: UserDetailsContextType) => {
+			return {
+				...prev,
+				govType
+			};
+		});
+	};
+
+	useEffect(() => {
+		if(network){
+			!isOpenGovSupported(network) && setGovTypeToContext(EGovType.GOV1);
+		}
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	},[network]);
+
+	const handleClick = () => {
+		if(!isOpenGovSupported(network)) return;
+		router.push(govType === EGovType.OPEN_GOV ? '/' : '/opengov' );
+		setGovTypeToContext(govType === EGovType.OPEN_GOV  ? EGovType.GOV1 : EGovType.OPEN_GOV);
+	};
+
 	return (
 		<Header className={`${className} shadow-md z-[1001] sticky top-0 flex items-center bg-white h-[60px] max-h-[60px] px-6 leading-normal border-solid border-t-0 border-r-0 border-b-2 border-l-0 border-pink_primary`}>
 			<MenuOutlined className='lg:hidden mr-5' onClick={() => {
 				setSidedrawer(!sidedrawer);
 			}} />
 			<nav className='w-full flex items-center justify-between h-[60px] max-h-[60px]'>
 				<div className='flex items-center'>
-					<Link className='flex' href={isGov2Route ? '/opengov' : '/'}><PaLogo className='w-[99px] h-[32px] md:w-[116px] md:h-[39px]' /></Link>
+					<div className='flex cursor-pointer' onClick={handleClick}>
```

Comment:
should be `Link` instead of div

Reply:
use e.stopPropagation() and e.preventDefault()

Reply:
if(!isOpenGovSupported(network)) return; because of this we can't make it Link